src/neo/SmartContract/ApplicationEngine.Binary.cs

```diff
@@ -16,5 +19,15 @@ internal StackItem BinaryDeserialize(byte[] data)
         {
             return BinarySerializer.Deserialize(data, MaxStackSize, MaxItemSize, ReferenceCounter);
         }
+
+        internal string Base64Encode(byte[] data)
+        {
+            return ToBase64String(data);
+        }
+
+        internal byte[] Base64Decode(string s)
+        {
+            return FromBase64String(s);
```

Comment:
Am I correct in that it will throw an exception and abort execution if the input string is not really base64?

Reply:
Yes, it will

Reply:
I'm thinking if there is a case when the contract wants to handle this error by itself. This will certainly get used for all sorts of inputs and maybe failing the TX completely won't be appropriate for some contracts. So catching the exception and returning something to the contract seems more useful to me.

Reply:
See https://github.com/neo-project/neo-vm/pull/339